### PR TITLE
Fix Issue #669 : Add Check for Module and Theme exists before uninstall.

### PR DESCRIPTION
**Pull request by bhanu951**
Resolves https://github.com/apigee/apigee-devportal-kickstart-drupal/issues/669

### CODE REVIEW

**Comment by divya-intelli** [comment]
@bhanu951 , thank you for the PR, we will review the PR.

File: apigee_devportal_kickstart.install [hunk 1]
```diff
@@ -659,17 +659,30 @@ function apigee_devportal_kickstart_update_9001() {
...
-  \Drupal::service('theme_installer')->install(['olivero']);
+  $theme_installer->install(['claro']);
+  $theme_installer->install(['olivero']);
   // Set the admin theme as claro.Theme seven is deprecated and removed from Drupal 10.
   $config = \Drupal::service('config.factory')->getEditable('system.theme');
   $config->set('admin', 'claro')->save();
   // Uninstall deprecated theme and module.
-  \Drupal::service('theme_installer')->uninstall(['seven', 'bartik']);
-  \Drupal::service('module_installer')->uninstall(['hal', 'color']);
+  foreach($deprecated_modules as $deprecated_module) {
```
Comment (divya-intelli) [inline apigee_devportal_kickstart.install:677]: @bhanu951 , please set value for `$deprecated_modules` and `$deprecated_themes`

Comment (bhanu951) [reply apigee_devportal_kickstart.install:677]: @divya-intelli Done!

Comment (divya-intelli) [reply apigee_devportal_kickstart.install:677]: @bhanu951 , I am getting error when running the update.

Comment (bhanu951) [reply apigee_devportal_kickstart.install:677]: @divya-intelli  what error are you getting ?

Comment (divya-intelli) [reply apigee_devportal_kickstart.install:677]: image

Comment (bhanu951) [reply apigee_devportal_kickstart.install:677]: Can you share me Drupal version before upgrade and the version you are tying to upgrade ?
Did you try the upgrade with drush. 
I deployed the same changes to multiple sites and they work fine.

Comment (bhanu951) [reply apigee_devportal_kickstart.install:677]: I believe this issue can be reproduced only if you are upgrading your site from `apigee_devportal_kickstart 2.1.0` version and below.

Comment (divya-intelli) [reply apigee_devportal_kickstart.install:677]: Drupal error log. image

File: apigee_devportal_kickstart.install [hunk 2]
```diff
@@ -659,17 +659,33 @@ function apigee_devportal_kickstart_update_9001() {
...
   $config->set('admin', 'claro')->save();
+
+  $deprecated_themes = ['seven', 'bartik'];
+  $deprecated_modules = ['hal', 'color'];
   // Uninstall deprecated theme and module.
-  \Drupal::service('theme_installer')->uninstall(['seven', 'bartik']);
-  \Drupal::service('module_installer')->uninstall(['hal', 'color']);
+  foreach($deprecated_modules as $deprecated_module) {
+    if($module_handler->moduleExists($deprecated_module)) {
+      $module_installer->uninstall($deprecated_module);
```
Comment (divya-intelli) [inline apigee_devportal_kickstart.install]:
```suggestion
      $module_installer->uninstall([$deprecated_module]);
```

File: apigee_devportal_kickstart.install [hunk 3]
```diff
@@ -659,17 +659,33 @@ function apigee_devportal_kickstart_update_9001() {
...
+    }
+  }
+  foreach($deprecated_themes as $deprecated_theme) {
+    if($theme_handler->themeExists($deprecated_theme)) {
+      $theme_installer->uninstall($deprecated_theme);
```
Comment (divya-intelli) [inline apigee_devportal_kickstart.install]:
```suggestion
      $theme_installer->uninstall([$deprecated_theme]);  
```

File: apigee_devportal_kickstart.install (hunk 1 above)
Comment (bhanu951) [reply apigee_devportal_kickstart.install:677]: Oh I see, seems I made error while pasting the code from my local repo. Your suggestions should fix the issue. Please try again with latest commits. Thanks.

**Review by divya-intelli — APPROVED** [review]
LGTM

**Pull request merged**